### PR TITLE
23141 update pbs checker

### PR DESCRIPTION
**Pull request by milljm**
Pylint happy (I think), add argparse to allow `-w --wait` blocking ability.

### CODE REVIEW

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: scripts/are_queued_jobs_finished.py [hunk 1]
```diff
@@ -8,45 +12,110 @@
...
+            try:
+                return json.load(pbs_job)
+            except json.decoder.JSONDecodeError:
+                sys.exit('Not a valid json file')
+    sys.exit(f'File not found: {job_file}')
 
-    def yieldJobsResultPath(self):
-        for k, v in self.__job_data.items():
-            yield k, v
+def qstat_results(job_id):
```
[the remark this answers is not in the document]
Comment (milljm) [reply scripts/are_queued_jobs_finished.py:32]: yeah... I am thinking it will at line 70

[the remark this answers is not in the document]
Comment (loganharbour) [reply scripts/are_queued_jobs_finished.py:32]: okay so we want a job from forever ago to be considered a finish?

[the remark this answers is not in the document]
Comment (milljm) [reply scripts/are_queued_jobs_finished.py:32]: I think I know where the confusion is coming from: If you check this target out, and expand on "Bison Examples", you'll see how this script is being used: https://civet.inl.gov/job/1301248/

We are using this script to try and limit the number of times we need to run `run_tests` down to only twice. One to launch jobs, and then a second time to create a finished report.

So a failing qstat command, while we don't know why it failed, its a finished job. The second time we run `./run_tests`, the heavier TestHarness process will correctly figure out the "why" of it, and report it as such.

**Comment by milljm** [comment]
I'd like to investigate the `--wait` argument a bit more. Allowing for an integer in seconds to wait, would be a nice addition.

**Review by loganharbour — APPROVED** [review]

**Pull request merged**